projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -653,13 +654,9 @@ export class IgxOverlayService implements OnDestroy {
                 //  if the click is on the element do not close this overlay
                 if (!info.elementRef.nativeElement.contains(target)) {
```

Comment:
you can now get `info.elementRef.nativeElement` and add it to the `excludedElements` array. This way we will check if given element contains the target at single place.
_Note_: if this is inside click we should exit the method immediately - call `return` as we do now if the click is on `info.elementRef.nativeElement`.